ultra/tests/test_evaluate.py

```diff
@@ -130,6 +118,21 @@ def run_experiment():
                     )
                     total_step += 1
                     state = next_state
+                evaluation_check(
+                    agent=agent,
+                    agent_id=AGENT_ID,
+                    episode=episode,
+                    eval_rate=10,
+                    eval_episodes=1,
+                    max_episode_steps=2,
+                    episode_count=episode_count,
+                    policy_class=policy_class,
+                    scenario_info=("00", "eval_test"),
+                    timestep_sec=0.1,
+                    headless=True,
+                    log_dir=log_dir,
+                )
+                episode_count += 1
```

Comment:
Why do we conduct the evaluation check after env.step() and agent.act()

Reply:
If we want training performance and evaluation to be performed at the same time, then evaluation has to wait unit after some episodes are over. The downside is we dont get to see evaluation at initialization of the agent.

Reply:
I see, sorry I think you talked about this yesterday but I may need a bit more clarification on why we need to perform the training and evaluation at the same?

Reply:
To measure generalization, we need to compare training and test performance. If we don't train and test at the same time, we won't be comparing the same agent because after training the agent's parameters will be updated.

Reply:
Okay, that does make sense. The agents who be on different states and comparison between the evaluation and training would contain some offset.